lib/bamboo/template.ex

```diff
@@ -0,0 +1,271 @@
+defmodule Bamboo.Template do
+  @moduledoc """
+  Render emails with layouts, view modules, and templates.
+
+  This module allows rendering emails with layouts and views. Pass an
+  atom (e.g. `:welcome_email`) as the template name to render both HTML and
+  plain text emails. Use a string if you only want to render one type, e.g.
+  `"welcome_email.text"` or `"welcome_email.html"`.
+
+  ## Examples
```

Comment:
Nitpick: What would you think of renaming this heading to something like "Usage Examples"? My thought is that `## Examples` is already used by Elixir for doctests. When I first saw this I was expecting doctests to follow. Could you change the heading hear to something else for consistency with that convention?

Reply:
I see what you mean, but I think I'd rather keep it this way. This is very, _very_, similar to our [`Bamboo.Phoenix`](https://hexdocs.pm/bamboo/Bamboo.Phoenix.html#content) docs (copied it from there and then modified it), and Bamboo.Phoenix uses the same `## Examples` language. And I think it's the same pattern we use elsewhere in Bamboo (e.g. [`Bamboo.Adapter` docs](https://hexdocs.pm/bamboo/Bamboo.Adapter.html#content)). So I'm happy leaving it as `## Examples` unless you feel strongly about it.